http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/NonLeoMessageSubscriber.scala

```diff
@@ -90,22 +104,38 @@ class NonLeoMessageSubscriber[F[_]: Timer](gkeAlg: GKEAlgebra[F],
         F.unit
       else
         for {
-          runtimeFromGoogle <- computeService.getInstance(msg.resource.labels.projectId,
+          runtimeFromGoogle <- computeService.getInstance(msg.googleProject,
                                                           msg.resource.labels.zone,
                                                           InstanceName(msg.resource.labels.instanceId.toString))
           // We mark the runtime as Deleted, and delete the instance.
           // If instance deletion fails for some reason, it will be cleaned up by resource-validator
           _ <- runtimeFromGoogle.traverse { instance =>
             for {
-              _ <- clusterQuery
-                .markDeleted(msg.resource.labels.projectId,
-                             RuntimeName(instance.getName),
-                             ctx.now,
-                             Some("cryptomining"))
+              runtimeInfo <- clusterQuery
+                .getActiveClusterRecordByName(msg.googleProject, RuntimeName(instance.getName))
                 .transaction
-              _ <- computeService.deleteInstance(msg.resource.labels.projectId,
-                                                 msg.resource.labels.zone,
-                                                 InstanceName(msg.resource.labels.instanceId.toString))
+              _ <- runtimeInfo.traverse { runtime =>
+                for {
+                  _ <- clusterQuery
+                    .markDeleted(msg.googleProject, RuntimeName(instance.getName), ctx.now, Some("cryptomining"))
+                    .transaction
+                  _ <- computeService.deleteInstance(msg.googleProject,
+                                                     msg.resource.labels.zone,
+                                                     InstanceName(msg.resource.labels.instanceId.toString))
+                  userSubjectId <- samDao.getUserSubjectId(runtime.auditInfo.creator, runtime.googleProject)
+                  _ <- userSubjectId.traverse { sid =>
+                    val byteString = ByteString.copyFromUtf8(CryptominingUserMessage(sid).asJson.noSpaces)
+
+                    val message = PubsubMessage
+                      .newBuilder()
+                      .setData(byteString)
+                      .putAttributes("traceId", ctx.traceId.asString)
+                      .putAttributes("cryptomining", "true")
+                      .build()
+                    publisher.publishNativeOne(message)
```

Comment:
So this will publish to the `leonardo-pubsub` topic, right? Will it cause any issues in `LeoPubsubMessageSubscriber` to receive these messages which it doesn't know how to handle?

Also will it ok for bard/etc to receive messages from `leonardo-pubsub? I guess it can filter based on the `cryptomining=true` attribute.

Reply:
yea, that was my original thought...But people preferred we publish to a dedicated topic, so I'm making it a separate topic (didn't realize the change wasn't pushed Tues)

Reply:
Cool.. separate topic sounds good 👍